src/components/CartCounter/CartCounter.tsx

```diff
@@ -1,34 +1,78 @@
+// ts-nocheck
 import { useState } from "react";
 
 import Grid from "@mui/material/Unstable_Grid2";
 import { IconButton } from "@mui/material";
 import { Add, Remove } from "@mui/icons-material";
 
+import { useCart, useCartDispatch } from "../../pages/CartPage/CartContext";
+import { ActionType } from "../../pages/CartPage/types";
+
 import styles from "./CartCounter.module.scss";
 
-const CartCounter = () => {
-  const [amount, setAmount] = useState(1);
-  const plusAmount = () => {
-    setAmount(amount + 1);
+const CartCounter = ({
+  id,
+  amount,
+  setAmountBeforeAddToCart,
+}: {
+  id: number;
+  amount: number;
+  setAmountBeforeAddToCart?: React.Dispatch<React.SetStateAction<number>>;
+}) => {
```

Comment:
It's better to create a separate interface and put it right above the component itself
```
interface CartCounterProps {
  id: number;
  amount: number;
  setAmountBeforeAddToCart?: React.Dispatch<React.SetStateAction<number>>;
}
```